locator/abstract_replication_strategy.cc

```diff
@@ -155,22 +160,34 @@ const tablet_aware_replication_strategy* abstract_replication_strategy::maybe_as
     return dynamic_cast<const tablet_aware_replication_strategy*>(this);
 }
 
-void replication_factor_data::parse(const sstring& rf) {
+size_t replication_factor_data::parse(const sstring& rf) {
     if (rf.empty() || std::any_of(rf.begin(), rf.end(), [] (char c) {return !isdigit(c);})) {
         throw exceptions::configuration_exception(
                 format("Replication factor must be numeric and non-negative, found '{}'", rf));
     }
     try {
-        _count = std::stol(rf);
+        return std::stol(rf);
```

Comment:
We can also support empty strings and return 0 in this case (for the empty rack-list case)